File: src/test/java/recipeapplication/utility/IngredientConverterTest.java
package recipeapplication.utility;

import org.junit.Test;

import static junit.framework.TestCase.assertEquals;
import static junit.framework.TestCase.assertNull;

public class IngredientConverterTest
{
    @Test
    public void shouldConvertGramToOunce()
    {
        String input = "500g of chicken thigh";
        String output = "17.6oz of chicken thigh";

        assertEquals(output, IngredientConverter.toImperial(input));
    }

    @Test
    public void shouldConvertOunceToGram()
    {
        String input = "Another 10oz of chicken thigh";
        String output = "Another 284g of chicken thigh";

        assertEquals(output, IngredientConverter.toMetric(input));
    }

    @Test
    public void shouldConvertKilogramToPound()
    {
        String input = "Chicken thigh 2.5kg";
        String output = "Chicken thigh 5.5lbs";

        assertEquals(output, IngredientConverter.toImperial(input));
    }

    @Test
    public void shouldConvertPoundToKilogram()
    {
        String input = "Chicken thigh 5lbs";
        String output = "Chicken thigh 2.3kg";

        assertEquals(output, IngredientConverter.toMetric(input));
    }

    @Test
    public void shouldNotEditValueIfNotInCorrectFormatInToMetric()
    {
        String input = "Chicken thigh fivelbs";
        String output = "Chicken thigh fivelbs";

        assertEquals(output, IngredientConverter.toMetric(input));
    }

    @Test
    public void shouldNotEditValueIfNotInCorrectFormatInToImperial()
    {
        String input = "Chicken thigh bkg";
        String output = "Chicken thigh bkg";

        assertEquals(output, IngredientConverter.toImperial(input));
    }

    @Test
    public void shouldReturnNullIfInputIsNullForToImperial()
    {
        assertNull(IngredientConverter.toImperial(null));
    }

    @Test
    public void shouldReturnNullIfInputIsNullForToMetric()
    {
        assertNull(IngredientConverter.toMetric(null));
    }
}

File: src/test/java/recipeapplication/utility/IngredientConverterTest.kt
package recipeapplication.utility

import junit.framework.TestCase
import org.junit.Test

class IngredientConverterTest {
    @Test
    fun `Should convert gram to ounce`() {
        val input = "500g of chicken thigh"
        val output = "17.6oz of chicken thigh"
        TestCase.assertEquals(output, IngredientConverter.toImperial(input))
    }

    @Test
    fun `Should convert ounce to gram`() {
        val input = "Another 10oz of chicken thigh"
        val output = "Another 284g of chicken thigh"
        TestCase.assertEquals(output, IngredientConverter.toMetric(input))
    }

    @Test
    fun `Should convert kilogram to pound`() {
        val input = "Chicken thigh 2.5kg"
        val output = "Chicken thigh 5.5lbs"
        TestCase.assertEquals(output, IngredientConverter.toImperial(input))
    }

    @Test
    fun `Should convert pound to kilogram`() {
        val input = "Chicken thigh 5lbs"
        val output = "Chicken thigh 2.3kg"
        TestCase.assertEquals(output, IngredientConverter.toMetric(input))
    }

    @Test
    fun `Should not edit value if not in correct format in toMetric`() {
        val input = "Chicken thigh fivelbs"
        val output = "Chicken thigh fivelbs"
        TestCase.assertEquals(output, IngredientConverter.toMetric(input))
    }

    @Test
    fun `Should not edit value if not in correct format in toImperial`() {
        val input = "Chicken thigh bkg"
        val output = "Chicken thigh bkg"
        TestCase.assertEquals(output, IngredientConverter.toImperial(input))
    }
}
